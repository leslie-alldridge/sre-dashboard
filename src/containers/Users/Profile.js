import React, { Component } from "react";
import axios from "axios";
import { Link, Redirect } from "react-router-dom";

import Table from "@material-ui/core/Table";
import Button from "@material-ui/core/Button";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableRow from "@material-ui/core/TableRow";

import {
  LinkButtons,
  deleteButton,
  updateButton,
  loginButton,
  logoutButton,
  HeaderBar,
  linkStyle,
  forgotButton
} from "../../components";

const loading = {
  margin: "1em",
  fontSize: "24px"
};

const title = {
  pageTitle: "User Profile Screen"
};

class Profile extends Component {
  constructor() {
    super();

    this.state = {
      first_name: "",
      last_name: "",
      email: "",
      username: "",
      password: "",
      isLoading: true,
      deleted: false,
      error: false
    };
  }

  async componentDidMount() {
    let accessString = localStorage.getItem("JWT");
    if (accessString == null) {
      this.setState({
        isLoading: false,
        error: true
      });
    } else {
      await axios
        .get("/findUser", {
          params: {
            username: this.props.match.params.username
          },
          headers: { Authorization: `JWT ${accessString}` }
        })
        .then(response => {
          this.setState({
            first_name: response.data.first_name,
            last_name: response.data.last_name,
            email: response.data.email,
            username: response.data.username,
            password: response.data.password,
            isLoading: false,
            error: false
          });
        })
        .catch(error => {
          console.log(error.data);
        });
    }
  }

  deleteUser = e => {
    let accessString = localStorage.getItem("JWT");
    if (accessString === null) {
      this.setState({
        loadingUser: false,
        error: true
      });
    }

    e.preventDefault();
    axios
      .delete("/deleteuser", {
        params: {
          username: this.props.match.params.username
        },
        headers: { Authorization: `JWT ${accessString}` }
      })
      .then(response => {
        localStorage.removeItem("JWT");
        this.setState({
          deleted: true
        });
      })
      .catch(error => {
        console.log(error.data);
      });
  };

  logout = e => {
    e.preventDefault();
    localStorage.removeItem("JWT");
  };

  render() {
    const {
      first_name,
      last_name,
      email,
      username,
      password,
      error,
      isLoading,
      deleted
    } = this.state;

    if (error) {
      return (
        <div>
          <HeaderBar title={title} />
          <div style={loading}>
            Problem fetching user data. Please login again.
          </div>
          <LinkButtons
            buttonText={`Login`}
            buttonStyle={loginButton}
            link={"/login"}
          />
        </div>
      );
    } else if (isLoading) {
      return (
        <div>
          <HeaderBar title={title} />
          <div style={loading}>Loading User Data...</div>
        </div>
      );
    } else if (deleted) {
      return <Redirect to="/" />;
    } else {
      return (
        <div>
          <HeaderBar title={title} />
          <Table>
            <TableBody>
              <TableRow>
                <TableCell>First Name</TableCell>
                <TableCell>{first_name}</TableCell>
              </TableRow>
              <TableRow>
                <TableCell>Last Name</TableCell>
                <TableCell>{last_name}</TableCell>
              </TableRow>
              <TableRow>
                <TableCell>Email</TableCell>
                <TableCell>{email}</TableCell>
              </TableRow>
              <TableRow>
                <TableCell>User Name</TableCell>
                <TableCell>{username}</TableCell>
              </TableRow>
              <TableRow>
                <TableCell>Password</TableCell>
                <TableCell style={{ WebkitTextSecurity: "disc" }}>
                  {password}
                </TableCell>
              </TableRow>
            </TableBody>
          </Table>
          <Button
            style={deleteButton}
            variant="contained"
            color="primary"
            onClick={this.deleteUser}
          >
            Delete User
          </Button>
          <LinkButtons
            buttonStyle={updateButton}
            buttonText={"Update User"}
            link={`/updateUser/${username}`}
          />
          <LinkButtons
            buttonStyle={forgotButton}
            buttonText={"Update Password"}
            link={`/updatePassword/${username}`}
          />
          <Button
            style={logoutButton}
            variant="contained"
            color="primary"
            onClick={this.logout}
          >
            <Link style={linkStyle} to={"/dashboard"}>
              Dashboard
            </Link>
          </Button>
          <Button
            style={logoutButton}
            variant="contained"
            color="primary"
            onClick={this.logout}
          >
            <Link style={linkStyle} to={"/"}>
              Logout
            </Link>
          </Button>
        </div>
      );
    }
  }
}

export default Profile;
