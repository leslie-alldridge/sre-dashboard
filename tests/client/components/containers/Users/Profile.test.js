import React from "react";
import { shallow } from "enzyme";
import Profile from "../../../../../src/containers/Users/Profile";
describe("MyComponent", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Profile debug />);

    expect(component).toMatchSnapshot();
  });
});
