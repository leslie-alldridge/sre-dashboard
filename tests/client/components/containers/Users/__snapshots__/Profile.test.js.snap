// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Profile
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <HeaderBar
          title={
            Object {
              "pageTitle": "User Profile Screen",
            }
          }
        />,
        <div
          style={
            Object {
              "fontSize": "24px",
              "margin": "1em",
            }
          }
        >
          Problem fetching user data. Please login again.
        </div>,
        <LinkButtons
          buttonStyle={
            Object {
              "background": "royalblue",
              "margin": "1em",
              "padding": "1em",
            }
          }
          buttonText="Login"
          link="/login"
        />,
      ],
      "id": "user-bg",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "title": Object {
            "pageTitle": "User Profile Screen",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Problem fetching user data. Please login again.",
          "style": Object {
            "fontSize": "24px",
            "margin": "1em",
          },
        },
        "ref": null,
        "rendered": "Problem fetching user data. Please login again.",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "buttonStyle": Object {
            "background": "royalblue",
            "margin": "1em",
            "padding": "1em",
          },
          "buttonText": "Login",
          "link": "/login",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HeaderBar
            title={
              Object {
                "pageTitle": "User Profile Screen",
              }
            }
          />,
          <div
            style={
              Object {
                "fontSize": "24px",
                "margin": "1em",
              }
            }
          >
            Problem fetching user data. Please login again.
          </div>,
          <LinkButtons
            buttonStyle={
              Object {
                "background": "royalblue",
                "margin": "1em",
                "padding": "1em",
              }
            }
            buttonText="Login"
            link="/login"
          />,
        ],
        "id": "user-bg",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "title": Object {
              "pageTitle": "User Profile Screen",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Problem fetching user data. Please login again.",
            "style": Object {
              "fontSize": "24px",
              "margin": "1em",
            },
          },
          "ref": null,
          "rendered": "Problem fetching user data. Please login again.",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "buttonStyle": Object {
              "background": "royalblue",
              "margin": "1em",
              "padding": "1em",
            },
            "buttonText": "Login",
            "link": "/login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
